libr/util/table.c

```diff
@@ -598,14 +598,98 @@ R_API void r_table_sortlen(RTable *t, int nth, bool dec) {
 	RTableColumn *col = r_list_get_n (t->cols, nth);
 	if (col) {
 		Gnth = nth;
-		Gcmp = cmplen;
 		t->rows->sorted = false; //force sorting
-		r_list_sort (t->rows, Gcmp);
+		r_list_sort (t->rows, cmplen);
 		if (dec) {
 			r_list_reverse (t->rows);
 		}
 		Gnth = 0;
-		Gcmp = NULL;
+	}
+}
+
+static int r_rows_cmp(RList *lhs, RList *rhs, RList *cols, int nth) {
+	RListIter *iter_lhs;
+	RListIter *iter_rhs;
+	RListIter *iter_col;
+	RTableColumn *item_col;
+
+	void *item_lhs;
+	void *item_rhs;
+	int tmp;
+	int i = 0;
+
+	for (iter_lhs = lhs->head, iter_rhs = rhs->head, iter_col = cols->head;
+		iter_lhs && iter_rhs && iter_col;
+		iter_lhs = iter_lhs->n, iter_rhs = iter_rhs->n, iter_col = iter_col->n) {
+
+		item_lhs = iter_lhs->data;
+		item_rhs = iter_rhs->data;
+		item_col = iter_col->data;
+
+		if (nth == -1 || i == nth) {
+			tmp = item_col->type->cmp (item_lhs, item_rhs);
+
+			if (tmp) {
+				return tmp;
+			}
+		}
+
+		++i;
+	}
+
+	if (iter_lhs) {
+		return 1;
+	}
+
+	if (iter_rhs) {
+		return -1;
+	}
+
+	return 0;
+}
+
+R_API void r_table_uniq(RTable *t) {
+	r_table_group (t, -1, NULL);
+}
+
+static void select_and_delete_the_row(RList *rows, RTableRow *uniq_row, RTableRow *row,
+	RListIter *iter_inner, RListIter *iter, RTableSelector fcn, int nth) {
+	int select_flag;
+	if (!fcn) {
+		r_list_delete (rows, iter);
+	} else {
+		select_flag = fcn (uniq_row->items, row->items, nth);
```

Comment:
why can `fcn` choose which element to delete? shouldn't it just merge the two rows together? I was thinking about a merge function that has an `accumulator` argument and a `new_row` argument.